src/gsi/pcgsoi.f90

```diff
@@ -400,14 +333,11 @@ subroutine pcgsoi()
         end if
      end if
 
-     if (iter==0 .and. print_diag_pcg) then
-        call prt_control_norms(grady,'grady')
-     endif
 
-!    Calculate new norm of gradients
-     if (iter>0) gsave=gnorm(3)
+!    2. Calculate new norm of gradients and factors going into b calculation
```

Comment:
Since line 322 is labeled as step `2. Multiply by background error`, should like 337 be relabeled step 3 with all subsequent steps (lines 384, 420, 434, 476,  incremented by one?

Reply:
Agreed.  Should I make changes?

Reply:
I think we should.  Mike, is there an easy way for John to directly change code in this PR?

Reply:
Thanks, Mike.   John, please make this change in the restart branch in your fork.  Mike will pull your update into his copy (and this PR).   Thank you, Mike. for getting John's update into this PR.

Reply:
Made these changes and committed resulting branch.  Will await further comments before making any other changes.